crates/loans/src/lib.rs

```diff
@@ -1295,9 +1295,11 @@ pub mod pallet {
             let from = Self::incentive_reward_account_id();
             Self::ensure_active_market(asset_id)?;
             Self::accrue_interest(asset_id)?;
-            let exchange_rate = Self::exchange_rate_stored(asset_id)?;
-            let voucher_amount = Self::calc_collateral_amount(redeem_amount, exchange_rate)?;
-            let redeem_amount = Self::do_redeem_voucher(&from, asset_id, voucher_amount)?;
+
+            let redeem_amount = Amount::new(redeem_amount, asset_id);
+            let voucher = redeem_amount.convert_to(Self::lend_token_id(asset_id)?)?;
```

Comment:
This is much cleaner, though one thing to mention is that `CurrencyConvert` tries to re-accrue interest. If interest has already been accrued in this block it's a no-op but I wasn't sure if there would be any edge cases caused by always re-accruing.

Reply:
Why would there by edge cases if it's a no-op?

Reply:
The side effect would be accruing interest more often, which is good for the lending protocol but will result in slightly higher fees